zio-http/src/main/scala/zhttp/service/Handler.scala

```diff
@@ -23,26 +23,33 @@ private[zhttp] final case class Handler[R](
   override def channelRead0(ctx: Ctx, jReq: FullHttpRequest): Unit = {
     jReq.touch("server.Handler-channelRead0")
     implicit val iCtx: ChannelHandlerContext = ctx
-    unsafeRun(
-      jReq,
-      app,
-      new Request {
-        override def method: Method = Method.fromHttpMethod(jReq.method())
-
-        override def url: URL = URL.fromString(jReq.uri()).getOrElse(null)
-
-        override def headers: Headers = Headers.make(jReq.headers())
-
-        override def remoteAddress: Option[InetAddress] = {
-          ctx.channel().remoteAddress() match {
-            case m: InetSocketAddress => Some(m.getAddress)
-            case _                    => None
-          }
-        }
-
-        override def data: HttpData = HttpData.fromByteBuf(jReq.content())
-      },
-    )
+    try
+      (
+        unsafeRun(
+          jReq,
+          app,
+          new Request {
+            override def method: Method = Method.fromHttpMethod(jReq.method())
+
+            override def url: URL = URL.fromString(jReq.uri()).getOrElse(null)
+
+            override def headers: Headers = Headers.make(jReq.headers())
+
+            override def remoteAddress: Option[InetAddress] = {
+              ctx.channel().remoteAddress() match {
+                case m: InetSocketAddress => Some(m.getAddress)
+                case _                    => None
+              }
+            }
+
+            override def data: HttpData = HttpData.fromByteBuf(jReq.content())
+          },
+        ),
+      )
+    catch {
+      case throwable: Throwable =>
+        writeResponse(Response.fromHttpError(HttpError.InternalServerError(cause = Some(throwable))), jReq): Unit
+    }
```

Comment:
- We should call the `Config.onError` handler if available
- If handler not available we should call `ctx.fireExceptionCaught`
- Connection is still not getting closed here?

Reply:
Connection is still not getting closed, Unless we write a 500 response or explicitly close the connection by `ctx.close()` . Also even if `Config.error` is available we can be sure that it will close the connection.

Reply:
Add a `connection:close` header. This will be understood automatically by the `Keep Alive` handle added to the channel pipeline.